csharp/src/Ice/IncomingResponseFrame.cs

```diff
@@ -165,13 +172,33 @@ public void ReadVoidReturnValue(IObjectPrx proxy)
             }
         }
 
+        /// <summary>Returns an <see cref="IncomingResponseFrame"/> that represents a oneway pseudo response.</summary>
+        internal static IncomingResponseFrame WithVoidReturnValue(Protocol protocol, Encoding encoding) =>
+            _cachedVoidReturnValueFrames.GetOrAdd((protocol, encoding), key =>
+            {
+                var data = new List<ArraySegment<byte>>();
+                var ostr = new OutputStream(key.Protocol.GetEncoding(), data);
+                ostr.Write(ResultType.Success);
+                _ = ostr.WriteEmptyEncapsulation(key.Encoding);
+                Debug.Assert(data.Count == 1);
+                return new IncomingResponseFrame(key.Protocol, data[0], int.MaxValue);
+            });
+
         /// <summary>Constructs an incoming response frame.</summary>
         /// <param name="protocol">The Ice protocol of this frame.</param>
         /// <param name="data">The frame data as an array segment.</param>
         /// <param name="maxSize">The maximum payload size, checked during decompress.</param>
-        public IncomingResponseFrame(Protocol protocol, ArraySegment<byte> data, int maxSize)
+        /// <param name="socketStream">The optional socket stream. The stream is non-null only if there's still data to
+        /// on the stream after the reading of the response frame.</param>
```

Comment:
` there's still data to on the stream after` this sounds weird

Reply:
Fixed